test/prefer-query-selector.js

```diff
@@ -0,0 +1,62 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/prefer-query-selector';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+ruleTester.run('prefer-query-selector', rule, {
+	valid: [
+		'document.querySelector("#foo");',
+		'document.querySelector("#foo .bar");',
+		'document.querySelector("main #foo .bar");'
+	],
+	invalid: [
+		{
+			code: 'document.getElementById("foo");',
+			errors: [{message: 'Prefer `querySelector` over `getElementById`.'}],
+			output: 'document.querySelector("#foo");'
+		},
+		{
+			code: 'document.querySelectorAll("p");',
+			errors: [{message: 'Prefer `querySelector` over `querySelectorAll`.'}],
+			output: 'document.querySelector("p");'
+		},
+		{
+			code: 'document.getElementsByClassName("foo");',
+			errors: [{message: 'Prefer `querySelector` over `getElementsByClassName`.'}],
+			output: 'document.querySelector(".foo");'
+		},
+		{
+			code: 'document.getElementsByClassName("foo bar");',
+			errors: [{message: 'Prefer `querySelector` over `getElementsByClassName`.'}],
+			output: 'document.querySelector(".foo.bar");'
+		},
+		{
+			code: 'document.getElementsByTagName("div");',
+			errors: [{message: 'Prefer `querySelector` over `getElementsByTagName`.'}],
+			output: 'document.querySelector("div");'
+		},
+		{
+			code: 'document.getElementById("foo").getElementsByClassName("bar baz");',
+			errors: [
+				{message: 'Prefer `querySelector` over `getElementsByClassName`.'},
+				{message: 'Prefer `querySelector` over `getElementById`.'}
+			],
+			output: 'document.querySelector("#foo").querySelector(".bar.baz");'
+		},
+		{
+			code: 'document.getElementById(\'foo\');',
+			errors: [{message: 'Prefer `querySelector` over `getElementById`.'}],
+			output: 'document.querySelector(\'#foo\');'
+		},
+		{
+			code: 'const qux = \'qux\'; document.getElementsByClassName(`foo ${qux}`);', // eslint-disable-line no-template-curly-in-string
+			errors: [{message: 'Prefer `querySelector` over `getElementsByClassName`.'}],
+			output: 'const qux = \'qux\'; document.querySelector(`.foo.${qux}`);' // eslint-disable-line no-template-curly-in-string
+		}
```

Comment:
There needs to be some more tests:

- When a variable is used: `getElementsByClassName(foo)`.
- When a variable is partly used: `getElementsByClassName(foo + 'bar')`

Reply:
We should allow usage like `getElementById(id)` (when the argument is a variable), as to not to encourage writing `querySelector('#' + id)`.

In other words: 
`getElementById(id)`
is better then
`querySelector('#' + CSS.escape(id))`
is better then
`querySelector('#' + id)`

Same goes for `getElementsByClassName` and `getElementsByTagName`.

Reply:
> We should allow usage like getElementById(id) (when the argument is not a string literal), as to not to encourage writing querySelector('#' + id).

I disagree. It's better to be consistent no matter what. It's an edge-case anyway, and `` querySelector(`#${id}`) `` is totally fine.

Why would you need to escape it with `querySelector` and not with `getElementById`?

Reply:
Why escape:
```js
const id = 'shoot foot';
const div = document.createElement('div');
div.id = id;
document.append(div);
document.querySelector(`#${CSS.escape(id)}`); // -> HTMLDivElement
document.querySelector(`#${id}`); // -> null
document.getElementById(id) // -> HTMLDivElement
```

Reply:
I have done web dev for 10+ years and have yet to see an ID or class name with spaces. I honestly don't think we should care about such an obscure edge-case.